controller/setting_controller.go

```diff
@@ -350,7 +350,9 @@ func (sc *SettingController) syncBackupTarget() (err error) {
 
 	existingBackupTarget := backupTarget.DeepCopy()
 	defer func() {
+		oldBackupTargetURL := backupTarget.Spec.BackupTargetURL
```

Comment:
Use `existingBackupTarget.Spec.BackupTargetURL` instead. 

Also, there is no need to create another variable, because you can do below. 

```go
if err = sc.handleSecretsForAWSIAMRoleAnnotation(backupTarget.Spec.BackupTargetURL, existingBackupTarget.Spec.CredentialSecret, secretSetting.Value, existingBackupTarget.Spec.BackupTargetURL != targetSetting.Value); err != nil {
				sc.logger.WithError(err).Warn("Failed to update secrets for AWSIAMRoleAnnotation")
			}
```

Reply:
Modified. Thanks.